### ISSUE TITLE
Support for scopes on relations

### ISSUE THREAD

**Issue by stayallive** [laravel-idea/plugin#366]
Hey there, I'm not a 100% sure this is fixable by a plugin but it would be extremely nice if it somehow could...

Considering this code:

```php
class Comment extends Model
{
    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class)->withTrashed();
    }
}
```

The `User` model has the `SoftDeletes` trait but PhpStorm generates this error:

```Method 'withTrashed' not found in \Illuminate\Database\Eloquent\Relations\BelongsTo```

Which is totally correct of course, however we all know this does work.

The nice thing is that with Larastan even PHPStan supports it ([PR](https://github.com/nunomaduro/larastan/pull/692)) and I thought that maybe there is a way to teach PhpStorm to handle this too so that autocompletion works correctly on relations.

If it's also possible to get this code working that would be even more cool:

```php
Comment::query()->first()->user()->withTrashed()->first();
```

I know this example doesn't make much sense but the gist is that it knows that the `user()` returns a relation that links to the user model so it knows `withTrashed` (or any other scope for that matter) is resolved.

If I missed something and this is already supported I'm sorry but I tried with the generated Eloquent code (which is crazy cool) and that did not seem to "fix" it.

Thanks in advance <3

**Comment by adelf** [comment]
![image](https://user-images.githubusercontent.com/2818394/129727717-72b3b07a-52b8-4a72-9762-23d54f0bbc9d.png)

The second one already works. PhpStorm here completes every method and correctly calculates the type for `$user` variable.

I'll try to add typing support for `belongsTo()` and other relation methods.

**Comment by adelf** [comment]
Forgot to say - you only need to generate an Eloquent helper code - https://laravel-idea.com/docs/eloquent

**Comment by stayallive** [comment]
Yes, you are absolutely right the second example does work already (pretty good actually), not sure what I missed there when I wrote this :)

The first would be really cool if it could be achieved too 💪

**Comment by gorbunov** [comment]
In first example this is not actually scope ->withTrashed(), but builder macro!

![Image](https://github.com/user-attachments/assets/6626052d-af23-44df-a683-86716112132a)

You can add yours builder macros as well, and they will be recognized!
